cli/command/stack/deploy.go

```diff
@@ -19,14 +31,15 @@ func newDeployCommand(dockerCli command.Cli) *cobra.Command {
 		Args:    cli.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			opts.Namespace = args[0]
+
 			if dockerCli.ClientInfo().HasKubernetes() {
 				kli, err := kubernetes.WrapCli(dockerCli, cmd)
 				if err != nil {
 					return err
 				}
-				return kubernetes.RunDeploy(kli, opts)
+				return kubernetes.RunDeploy(kli, opts, loadComposefile)
```

Comment:
I don't like this functor. I think the only reason is to avoid circular dependency between stack package and swarm/kubernetes packages. A more cleaner way could be moving the loader in its own package, like cli/command/stack/loader.